### PR TITLE
Add aliyun ecs instance join leave security group

### PR DESCRIPTION
**Pull request by SnailJie**
### Description

When I use libcloud in my project, I need use function about "Join/Leave Security Group" for Aliyun ECS instance. But I can not find it in libcloud. That's why I make this pull request.

Two functions are added: join_security_group/leave_security_group 

### Status

- done, ready for review

### Checklist (tick everything that applies)

- [x] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [x] Documentation
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [ ] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

**Comment by wsbmilk** [comment]
+10086

File: libcloud/compute/drivers/ecs.py [hunk 1]
```diff
@@ -872,6 +872,62 @@ def ex_list_security_group_attributes(self, group_id=None,
...
+        :type ex_filters: ``str``
+
+
+        :return: join operation result.
+        :rtype: ``bool``
+        """
+        if group_id is None:
+            raise AttributeError('group_id is required')
+
+        if (node.state != NodeState.RUNNING) and \
```
Comment (tonybaloney) [inline libcloud/compute/drivers/ecs.py]: no parentheses in Python :-) please use a more Pythonic evaluation statement.

File: libcloud/compute/drivers/ecs.py [hunk 2]
```diff
@@ -872,6 +872,62 @@ def ex_list_security_group_attributes(self, group_id=None,
...
+
+    def ex_leave_security_group(self, node, group_id=None):
+        """
+        Leave a node from security group.
+
+        :param node: The node to leave security group
+        :type node: :class:`Node`
+
+        :param group_id: security group id.
+        :type ex_filters: ``str``
```
Comment (tonybaloney) [inline libcloud/compute/drivers/ecs.py]: this should be `:type group_id:`

File: libcloud/compute/drivers/ecs.py [hunk 3]
```diff
@@ -872,6 +872,62 @@ def ex_list_security_group_attributes(self, group_id=None,
...
+
+
+        :return: leave operation result.
+        :rtype: ``bool``
+        """
+        if group_id is None:
+            raise AttributeError('group_id is required')
+
+        if (node.state != NodeState.RUNNING) and \
```
Comment (tonybaloney) [inline libcloud/compute/drivers/ecs.py]: again, no parens in Python

File: libcloud/compute/drivers/ecs.py [hunk 4]
```diff
@@ -872,6 +872,62 @@ def ex_list_security_group_attributes(self, group_id=None,
...
 
+    def ex_join_security_group(self, node, group_id=None):
+        """
+        Join a node into security group.
+
+        :param node: The node to join security group
+        :type node: :class:`Node`
+
+        :param group_id: security group id.
+        :type ex_filters: ``str``
```
Comment (tonybaloney) [inline libcloud/compute/drivers/ecs.py]: this docstring title is wrong, it should be `:type group_id`

**Review by tonybaloney — CHANGES_REQUESTED** [review]
Thanks for the contribution, please follow the changes requested in the code review.

File: libcloud/compute/drivers/ecs.py (hunk 4 above)
Comment (SnailJie) [reply libcloud/compute/drivers/ecs.py]: OK, Thank you, I will fix it

**Comment by tonybaloney** [comment]
thanks, merging.

**Comment by tonybaloney** [comment]
this is merged. I don't know why this is still here.

**Pull request merged**